app/controllers/posts_controller.rb

```diff
@@ -106,24 +108,28 @@ def new
 
   def create
     import_thread and return if params[:button_import].present?
-    preview and return if params[:button_preview].present?
 
     @post = current_user.posts.new(post_params)
+    @post.written = @post.replies.new(written_params)
+    @post.written.user = current_user
+
+    preview and return if params[:button_preview].present?
     @post.settings = process_tags(Setting, :post, :setting_ids)
     @post.content_warnings = process_tags(ContentWarning, :post, :content_warning_ids)
     @post.labels = process_tags(Label, :post, :label_ids)
 
-    unless @post.save
+    begin
+      @post.save!
+      flash[:success] = "You have successfully posted."
+      redirect_to post_path(@post)
+    rescue ActiveRecord::RecordInvalid
       flash.now[:error] = {}
       flash.now[:error][:array] = @post.errors.full_messages
       flash.now[:error][:message] = "Your post could not be saved because of the following problems:"
```

Comment:
Did we check this includes reply validation errors?

Reply:
We don't currently have a way to get an invalid reply in this context, since the user will always be the same as the post user and that's all we currently validate! I'll add some code to make sure we get reply errors if any exist though.